[Feature] Card component

aleksik opened the pull request:
## Description

This PR implements the Card component.

## How Has This Been Tested?

Tested in Storybook locally.

Live demo:
https://aleksik.github.io/core/?path=/story/components-card--empty
 https://aleksik.github.io/react/?path=/docs/components-card--empty

ronijaakkola commented:
Documentation done here: https://github.com/City-of-Helsinki/helsinki-design-system/pull/246
It should be merged into this branch and finalize it (add examples to Playground).

ronijaakkola reviewed (changes requested):
![image](https://user-images.githubusercontent.com/4214671/96085198-23657880-0ec9-11eb-8559-4156167be22a.png)

Spacing between elements seems a bit off. I just realized Abstract had a hard time displaying those in the showcase artboard inspect. So here are those in written format:

- Spacing between the title and text should be 24px (`--spacing-m` token)
- Spacing between the text and the button should be 32px (`--spacing-l` token)

---

Before merging, we should merge documentation branch here and finalize it (add examples to Playground). Documentation is done here: #246

aleksik commented:
@ronijaakkola spacing is now updated based on Abstract artboard. These fixes can be tested here:
https://aleksik.github.io/core/?path=/story/components-card--empty
https://aleksik.github.io/react/?path=/docs/components-card--empty

I also cherry-picked the documentation commits and added usage examples.

ronijaakkola reviewed (approved):
Hyvältä näyttää. Tein pari lisäystä dokumentaatioon ja korjasin jotain saavutettavuusongelmia (kun dokumentaatio oli ajalta ennen korjauksia).

niglu1 commented on packages/react/src/components/card/Card.module.css:
```diff
@@ -0,0 +1,19 @@
+.card {
```
Could this file be changed from `.css` to `.scss`?

niglu1 commented on packages/react/src/components/card/Card.stories.tsx:
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
```
Could the stories be written using `args`?

niglu1 reviewed (changes requested):
Nais 👌
